### PR TITLE
Add TypeKind::HUGEINT to Type dispatch templates

### PR DESCRIPTION
**Pull request by majetideepak**
Add TypeKind::HUGEINT support in
`VELOX_DYNAMIC_SCALAR_TEMPLATE_TYPE_DISPATCH`
`VELOX_DYNAMIC_TYPE_DISPATCH_IMPL`
`VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH`

### CODE REVIEW

**Comment by majetideepak** [comment]
Need https://github.com/facebookincubator/velox/issues/4864 for the `linux-adapters` CI to pass.

File: scripts/setup-macos.sh [hunk 1]
```diff
@@ -94,7 +94,7 @@ function install_fmt {
 function install_folly {
   github_checkout facebook/folly "v2022.11.14.00"
   OPENSSL_ROOT_DIR=$(brew --prefix openssl@1.1) \
-    cmake_install -DBUILD_TESTS=OFF
+  cmake_install -DBUILD_TESTS=OFF -DFOLLY_HAVE_INT128_T=ON
```
Comment (assignUser) [inline scripts/setup-macos.sh:97]: This needs to be added to the other install scripts as well!

Comment (majetideepak) [reply scripts/setup-macos.sh:97]: done. Thanks!

Comment (assignUser) [reply scripts/setup-macos.sh:97]: The linux wheels (and any not running the script during the run) will fail as long as the docker image is not rebuild

Comment (majetideepak) [reply scripts/setup-macos.sh:97]: I opened https://github.com/facebookincubator/velox/issues/4864 to rebuild the docker image.

**Comment by kgpai** [comment]
@majetideepak The failure in pyVelox is because I think it uses bundled folly without hugeint support. Might need a change in setup.py:build_extension to pass right flags to fix this. 
cc @assignUser

**Comment by majetideepak** [comment]
@kgpai I likely did not rebase this branch after the script changes. I pushed again. Let's see if that fixes the CI job.

**Comment by majetideepak** [comment]
@kgpai it now passes. Can you please help review/merge?

**Review by kgpai — APPROVED** [review]

**Comment by kgpai** [comment]
@majetideepak Please give me a day or two to merge this as it requires some changes internally.

**Pull request closed without merging**